src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -259,6 +344,16 @@
     private ScrollingList userList;
 
     /**
+     * The ScrollingList of the users.
+     */
+    private ScrollingList actionsList;
+
+    /**
+     * The ScrollingList of the users.
+     */
+    private ScrollingList freeBlocksList;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove the "freeBlocksList" field and declare it as a local variable in the relevant methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1450)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Perhaps 'freeBlocksList' could be replaced by a local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ASingularField)